src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs

```diff
@@ -243,7 +243,10 @@ public SslProtocols SslProtocols
                 _underlyingHandler.SendAsync(request, cancellationToken);
         }
 
-        public static Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool> DangerousAcceptAnyServerCertificateValidator { get; } = delegate { return true; };
+        // lazy-load the validator func so it can be trimmed by the ILLinker if it isn't used.
+        private static Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool>? s_dangerousAcceptAnyServerCertificateValidator;
+        public static Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool> DangerousAcceptAnyServerCertificateValidator =>
+            s_dangerousAcceptAnyServerCertificateValidator ??= delegate { return true; };
```

Comment:
While we don't currently in SocketsHttpHandler, in CurlHandler we relied on object identity to determine whether the supplied handler was this one, letting us skip the relevant validation logic if this instance was supplied.  I think we need to retain that.  That means this needs to be made thread-safe such that it's idempotent even when accessed concurrently, e.g.
```C#
public static Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool> DangerousAcceptAnyServerCertificateValidator =>
    Volatile.Read(ref s_dangerousAcceptAnyServerCertificateValidator) ??
    Interlocked.CompareExchange(s_dangerousAcceptAnyServerCertificateValidator, delegate { return true; }, null) ??
    s_dangerousAcceptAnyServerCertificateValidator;
```

Reply:
Thanks.  In general I think we need to default to assuming that objects exposed from static properties have such a requirement.